src/tools/clippy/clippy_lints/src/loops/needless_range_loop.rs

```diff
@@ -256,12 +262,15 @@ impl<'a, 'tcx> VarVisitor<'a, 'tcx> {
             match res {
                 Res::Local(hir_id) => {
                     let parent_def_id = self.cx.tcx.hir().get_parent_item(expr.hir_id);
-                    let extent = self
-                        .cx
-                        .tcx
-                        .region_scope_tree(parent_def_id)
-                        .var_scope(hir_id.local_id)
-                        .unwrap();
+                    let region_scope_tree = self.cx.tcx.region_scope_tree(parent_def_id);
+                    let scope_map = (self.cx.tcx.sess.at_least_rust_2024() && self.cx.tcx.features().new_temp_lifetime)
+                        .then(|| self.cx.tcx.body_scope_map(parent_def_id));
+                    let extent = if let Some(scope_map) = scope_map {
+                        scope_map.var_scope_new(hir_id.local_id)
```

Comment:
This implementation differs from the implementation 200 lines above.
```suggestion
                    let extent = if self.cx.tcx.sess.at_least_rust_2024() && self.cx.tcx.features().new_temp_lifetime {
                        let scope_map = self.cx.tcx.body_scope_map(parent_def_id))
                        scope_map.var_scope_new(hir_id.local_id)
```

Reply:
Applied